### PR TITLE
Added French/German/Spanish

### PR DESCRIPTION
**Pull request by shivjyotigarai**
added the above languages to make it more usable across the globe..

### CODE REVIEW

File: .idea/compiler.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (iamsh4shank) [inline .idea/compiler.xml:1]: No need of this

**Review by iamsh4shank — COMMENTED** [review]
Hey @shivjyotigarai this PR is for the language translations, can you add a settings page (preferences page)and add an option to switch languages ?

File: .idea/compiler.xml (hunk 1 above)
Comment (shivjyotigarai) [reply .idea/compiler.xml:1]: yeah...i am just adding a button that will allow the user to choose the above languages...will that do?

**Comment by shivjyotigarai** [comment]
Yeah

File: .idea/compiler.xml (hunk 1 above)
Comment (iamsh4shank) [reply .idea/compiler.xml:1]: yeah, you can add a preference screen for settings that will provide you a feature called switch preferences to switch between various languages. For implementing this you can search what a preference setting is.

Comment (shivjyotigarai) [reply .idea/compiler.xml:1]: yeah I tried using the preference setting screen...but it opens the entire list of languages available on android....but the app supports only 4 languages....so i made a button on top right ...such that the user can switch between the available languages....would you plz look at it once ...attaching the screenshots

Comment (shivjyotigarai) [reply .idea/compiler.xml:1]: image
image
image

Comment (shivjyotigarai) [reply .idea/compiler.xml:1]: Added a new commit...plz look at it once :)

Comment (shivjyotigarai) [reply .idea/compiler.xml:1]: Oops!! Forgot to translate the choose language text

Comment (shivjyotigarai) [reply .idea/compiler.xml:1]: Now i think it should work fine :)

File: app/src/main/res/values-de-rDE/strings.xml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (shivjyotigarai) [inline app/src/main/res/values-de-rDE/strings.xml:1]: should i remove this line of code :)

File: .idea/compiler.xml (hunk 1 above)
Comment (shivjyotigarai) [reply .idea/compiler.xml:1]: > No need of this

Should I remove that line😊

**Comment by iamsh4shank** [comment]
Hey @shivjyotigarai can you put this choose a language thing inside the settings page, as we know currently the app doesn't have the settings page so can you put it in the app title bar? For example, there is a logout button, so just put it to the left of the logout button for now

**Comment by shivjyotigarai** [comment]
where is the logout button...could you plz mention the activity name for the same 👍

**Comment by iamsh4shank** [comment]
If you open the app so on the top right side there is a button in the shape of a gate, that's the logout button, so it would be great if you put the language change button there.

**Comment by shivjyotigarai** [comment]
understood..ok working on it

**Comment by iamsh4shank** [comment]
Cool just add a similar language button there to switch between the languages.

File: app/src/main/java/com/example/healthlog/MainActivity.java [hunk 3]
```diff
@@ -6,6 +6,8 @@
 import android.view.Menu;
 import android.view.MenuInflater;
 import android.view.MenuItem;
+
```
Comment (iamsh4shank) [inline app/src/main/java/com/example/healthlog/MainActivity.java:9]: No need of this space

File: app/src/main/java/com/example/healthlog/MainActivity.java [hunk 4]
```diff
@@ -34,6 +36,13 @@ protected void onCreate(Bundle savedInstanceState) {
         NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment);
         NavigationUI.setupActionBarWithNavController(this, navController, appBarConfiguration);
         NavigationUI.setupWithNavController(navView, navController);
+
```
Comment (iamsh4shank) [inline app/src/main/java/com/example/healthlog/MainActivity.java:39]: here too

**Review by iamsh4shank — CHANGES_REQUESTED** [review]
Also, try to add shared preferences to save the language, so that in the future if the user restarts the app then the language will be retained from previous settings.

**Comment by shivjyotigarai** [comment]
> Also, try to add shared preferences to save the language, so that in the future if the user restarts the app then the language will be retained from previous settings.

yeah i have added already....

**Comment by shivjyotigarai** [comment]
@robustTechie just saw that the logout button is there on the top right after someone is logged in....not when the app first opens...so the user would have to enter via english only....like the **hospital id text**, **the id itself** and the **Login button text**  would be available only in english...**if the CHOOSE LANGUAGE button is available just after the app  is opened (i.e where its presently now) for the first time (i.e before logging in ) then the user can enter in his respective language from the very beginning**  What do you think about it ? :)

**Comment by iamsh4shank** [comment]
You can make it in that way too but that click button looks weird in terms of the position in the app, so one suggestion if you want to provide the feature from the login time then provide it as a button same as logout i.e. in the top right of the app bar.

Let me know if you have any doubt :)

**Comment by shivjyotigarai** [comment]
> You can make it in that way too but that click button looks weird in terms of the position in the app, so one suggestion if you want to provide the feature from the login time then provide it as a button same as logout i.e. in the top right of the app bar.
> 
> Let me know if you have any doubt :)

yeah...its looking quite weird....working on it...if i face any problem i will let u know :)

**Comment by shivjyotigarai** [comment]
Now fine ?? 😊😊

**Comment by iamsh4shank** [comment]
> Now fine ??

Not their just opposite to the Health Log in the title bar

**Comment by iamsh4shank** [comment]
Also, use some icon which is significant to language

**Comment by shivjyotigarai** [comment]
Ok

**Comment by shivjyotigarai** [comment]
Wrongly committed...the last one...fixing that

**Comment by shivjyotigarai** [comment]
@robustTechie  would you plz tell me how to customize the action bar...and add a button there....I moved to the styles.xml file and there changed **<style name="AppTheme" parent="Theme.AppCompat.Light.DarkActionBar">** to **<style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">**   and then in the LoginActivity tried to add my own Title Bar and a button there ...BUT THE APP CRASHED....plz tell me what to do :)

**Comment by iamsh4shank** [comment]
Hey, @shivjyotigarai don't change it to no action bar, because of this the app is crashed as you removed the action bar, you can add the icon by accessing the action bar in the app and then adding the icon there.

**Comment by shivjyotigarai** [comment]
> Hey, @shivjyotigarai don't change it to no action bar, because of this the app is crashed as you removed the action bar, you can add the icon by accessing the action bar in the app and then adding the icon there.

will you plz mention the xml file name specifically where can i access bar...and add a button there

**Comment by iamsh4shank** [comment]
I would suggest you to shift this choose language feature after the user logins

**Comment by shivjyotigarai** [comment]
> I would suggest you to shift this choose language feature after the user logins

plz tell me the layout file name ...i would add the add the button there :)

**Comment by shivjyotigarai** [comment]
image
in which layout should i add the button....

**Comment by shivjyotigarai** [comment]
image
Like here in the doctors activity the action bar is accessed , but where the logout button is added..that opens an alert msg prompting the user to logout....

**Comment by shivjyotigarai** [comment]
acha got it...it is present in menu resource folder

**Comment by shivjyotigarai** [comment]
I have fixed the issue completely and it is working all fine.....attaching a few pics of the same :)

**Comment by shivjyotigarai** [comment]
Is everything fine now ??😊😊

**Comment by shivjyotigarai** [comment]
Plz look at it once 😊😊

**Comment by robustTechie** [comment]
Hey @shivjyotigarai looks fine to me, did you added the locale to the shared preferences? Also please squash the commits :)

File: .idea/jarRepositories.xml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (iamsh4shank) [inline .idea/jarRepositories.xml:1]: no need of this

File: .idea/misc.xml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (iamsh4shank) [inline .idea/misc.xml:1]: no need to change this

File: app/src/main/java/com/example/healthlog/LoginActivity.java [hunk 7]
```diff
@@ -35,7 +53,103 @@ public void onClick(View v) {
                         signIn();
                     }
                 });
+
```
Comment (iamsh4shank) [inline app/src/main/java/com/example/healthlog/LoginActivity.java:52]: don't add extra lines with no use

**Comment by shivjyotigarai** [comment]
@robustTechie  Yeah , I have added shared preferences...you can see the code :).......yes squashing the commits

**Comment by shivjyotigarai** [comment]
@robustTechie squashed all the commits into 1...you can merge/approve the pull request now :)

**Review by iamsh4shank — CHANGES_REQUESTED** [review]
See my comments, else everything looks nice to me. Can you also add your local to the shared preference to save the language

**Comment by shivjyotigarai** [comment]
> See my comments, else everything looks nice to me. Can you also add your local to the shared preference to save the language 

Means hindi?

**Comment by iamsh4shank** [comment]
> > See my comments, else everything looks nice to me. Can you also add your local to the shared preference to save the language
> 
> Means hindi?

No I mean you have a locale variable right, so add it to shared preferences. You can read about the shared preferences so that whenever you close the app and then again restart the app then it will retain the last selected languages. Did you get it ?

**Comment by shivjyotigarai** [comment]
> > > See my comments, else everything looks nice to me. Can you also add your local to the shared preference to save the language
> > 
> > 
> > Means hindi?
> 
> No I mean you have a locale variable right, so add it to shared preferences. You can read about the shared preferences so that whenever you close the app and then again restart the app then it will retain the last selected languages. Did you get it ?

Yes I have added shared preferences from the very beginning....when the app is closed and again started then it restarts with the last selected languages....checked it many times :)

**Comment by iamsh4shank** [comment]
Hey @shivjyotigarai  for now do one thing just remove that compiler.xml file then I will merge it. And I will raise a new issue to maintain shared preferences then you can add the changes there. Fine?

**Comment by shivjyotigarai** [comment]
> Hey @shivjyotigarai for now do one thing just remove that compiler.xml file then I will merge it. And I will raise a new issue to maintain shared preferences then you can add the changes there. Fine?

yeah..deleted those files as u have mentioned :)

File: .idea/gradle.xml [hunk 8]
```diff
@@ -14,6 +14,7 @@
           </set>
         </option>
         <option name="resolveModulePerSourceSet" value="false" />
+        <option name="useQualifiedModuleNames" value="true" />
```
Comment (iamsh4shank) [inline .idea/gradle.xml:17]: don't do these unwanted changes :)

**Review by iamsh4shank — CHANGES_REQUESTED** [review]
Don't delete .idea file completely, what I wanted to say is don't make any changes in that file

File: .idea/gradle.xml (hunk 8 above)
Comment (iamsh4shank) [reply .idea/gradle.xml:17]: Also squash your commits

Comment (shivjyotigarai) [reply .idea/gradle.xml:17]: ok

Comment (shivjyotigarai) [reply .idea/gradle.xml:17]: > Also squash your commits

can u plz squash those commits..i am facing a server error from my side :)

**Comment by shivjyotigarai** [comment]
@robustTechie  is everything fine?

**Comment by shivjyotigarai** [comment]
image
I cant understand why these commits are comming while squashing..... @robustTechie plz tell what to do :)

**Review by iamsh4shank — COMMENTED** [review]
> image
> I cant understand why these commits are comming while squashing..... @robustTechie plz tell what to do :)

remove pick in each line and then put **r**  in first 9 lines in the place of pick and **f** in 10th line and then save it

**Comment by shivjyotigarai** [comment]
> > image
> > I cant understand why these commits are comming while squashing..... @robustTechie plz tell what to do :)
> 
> remove pick in each line and then put **r** in first 9 lines in the place of pick and **f** in 10th line and then save it

image
then ? @robustTechie

**Comment by iamsh4shank** [comment]
again save it

**Comment by shivjyotigarai** [comment]
image
after saving @robustTechie

**Comment by iamsh4shank** [comment]
Save it

**Comment by shivjyotigarai** [comment]
> Save it

i cant..after saving using **esc + : + wq** the same msg prompts

**Comment by shivjyotigarai** [comment]
> Save it

BTW, what are all this stuff going on 😅😅

**Comment by iamsh4shank** [comment]
How many commit did you had previously?

**Comment by shivjyotigarai** [comment]
> How many commit did you had previously?

6

**Comment by shivjyotigarai** [comment]
> How many commit did you had previously?

but they have been squashed...these commits are not made by me

**Comment by shivjyotigarai** [comment]
But @robustTechie the pull request can be merged without squashing 🧐🤔

**Comment by shivjyotigarai** [comment]
if any problem persists, then i can generate a new pull request regarding the same...i will copy paste the code in a new file :) @robustTechie

**Comment by iamsh4shank** [comment]
for squashing 6 commits you need to write git rebase -i HEAD~6

**Comment by shivjyotigarai** [comment]
> for squashing 6 commits you need to write git rebase -i HEAD~6

those commits are squashed ... i am refering to the 5 commits present now...the deleted files commits...can u just squash them from ur side :)

**Comment by iamsh4shank** [comment]
A good practice is to first squash commits and then push, can you once send the log of the git

**Comment by shivjyotigarai** [comment]
> A good practice is to first squash commits and then push, can you once send the log of the git

image
image

**Comment by shivjyotigarai** [comment]
like the update Readme and UI enhacements are not done by me :)

**Comment by iamsh4shank** [comment]
write `git rebase -i HEAD~3` and then remove pick and put r in first two and f in third one then save it

**Comment by shivjyotigarai** [comment]
> write `git rebase -i HEAD~3` and then remove pick and put r in first two and f in third one then save it

can i write squash instead of r and f...will that do the same thing?

**Comment by shivjyotigarai** [comment]
> write `git rebase -i HEAD~3` and then remove pick and put r in first two and f in third one then save it

image

**Comment by iamsh4shank** [comment]
write `git rebase --abort` then `git rebase -i HEAD~3`

**Comment by shivjyotigarai** [comment]
`fatal: No rebase in progress?`

**Comment by iamsh4shank** [comment]
did you changed something in your code that's why it is showing you need to commit first

**Comment by shivjyotigarai** [comment]
> did you changed something in your code that's why it is showing you need to commit first

no....i will create a new pull request tomorrow ..it has become very messy...what about it

**Comment by iamsh4shank** [comment]
once check ig you made some changes in your code so first commit them then rebase

**Comment by shivjyotigarai** [comment]
> once check ig you made some changes in your code so first commit them then rebase

no nothing is changed..seen

**Comment by iamsh4shank** [comment]
I don't think so their must be some changes in any file maybe jdk version change try by adding it and then using diff

**Comment by shivjyotigarai** [comment]
> I don't think so their must be some changes in any file maybe jdk version change try by adding it and then using diff

image
these 2 files are always there

**Comment by iamsh4shank** [comment]
yeah that's why commit it then do `git rebase -i HEAD~4`

**Comment by shivjyotigarai** [comment]
> git rebase -i HEAD~4

will it be 4?

**Comment by shivjyotigarai** [comment]
> yeah that's why commit it then do `git rebase -i HEAD~4`

r in the first 3 and f in the last?

**Comment by iamsh4shank** [comment]
yes

**Comment by shivjyotigarai** [comment]
`Successfully rebased and updated refs/heads/fix-language.`
Done finally...

**Comment by iamsh4shank** [comment]
cool now push it

**Comment by shivjyotigarai** [comment]
Done !!!!

**Comment by iamsh4shank** [comment]
Can you again send me the log

**Comment by shivjyotigarai** [comment]
> Can you again send me the log
image

image

**Comment by iamsh4shank** [comment]
You still have three comits which are needed to be squashed, put `git rebase -i HEAD~3`

**Comment by shivjyotigarai** [comment]
done !!  @robustTechie

**Comment by shivjyotigarai** [comment]
image
still 2 r left i think

**Comment by iamsh4shank** [comment]
again two commits :P `git rebase -i HEAD~2`
put r in first and f in last

**Comment by shivjyotigarai** [comment]
@robustTechie I was facing some issues with my laptop yesterday..many files got deleted including the project itself...So I have re-written the code again and generating a new pull request for the same...Very sorry for the inconvenience caused

**Pull request closed without merging**